File: sdk-python/examples/subscriptions/manage_url.py
"""Run with: python examples/subscriptions/manage_url.py"""

from __future__ import annotations

from threecommon import ThreeCommon


def main() -> None:
    with ThreeCommon(api_key="3co_your_api_key_here") as client:
        portal = client.subscriptions.retrieve_manage_url("sub_replace_with_real_id")
        print(f"manage URL: {portal.url}")


if __name__ == "__main__":
    main()

File: sdk-node/examples/subscriptions/manage-url.ts
/**
 * Fetch the signed self-service portal URL for a subscription. Share the
 * returned link with the subscriber so they can view, cancel, or resume it.
 *
 * Run:
 *   npx tsx examples/subscriptions/manage-url.ts
 */

import { ThreeCommon } from '@3common/sdk'

const client = new ThreeCommon({
  apiKey: '3co_your_api_key_here',
})

const { url } = await client.subscriptions.retrieveManageUrl('sub_replace_with_real_id')

console.log(`manage URL: ${url}`)
